pkg/util/parameters.go

```diff
@@ -0,0 +1,180 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	ackerr "github.com/aws-controllers-k8s/runtime/pkg/errors"
+)
+
+var (
+	errUnknownParameter      = fmt.Errorf("unknown parameter")
+	errUnmodifiableParameter = fmt.Errorf("parameter is not modifiable")
+)
+
+func NewErrUnknownParameter(name string) error {
+	// This is a terminal error because unless the user removes this parameter
+	// from their list of parameter overrides, we will not be able to get the
+	// resource into a synced state.
+	return ackerr.NewTerminalError(
+		fmt.Errorf("%w: %s", errUnknownParameter, name),
+	)
+}
+
+func NewErrUnmodifiableParameter(name string) error {
+	// This is a terminal error because unless the user removes this parameter
+	// from their list of parameter overrides, we will not be able to get the
+	// resource into a synced state.
+	return ackerr.NewTerminalError(
+		fmt.Errorf("%w: %s", errUnmodifiableParameter, name),
+	)
+}
+
+// ComputeParametersDelta compares two Parameter arrays and returns the new
+// parameters to add, to update and the parameter identifiers to delete
+func ComputeParametersDelta(
```

Comment:
Check out how I did this for tags in the runtime using `lo` helper methods - https://github.com/aws-controllers-k8s/runtime/blob/main/pkg/compare/tags.go#L26

It might not be any shorter, but I think the code is more maintainable

Reply:
Done